### ISSUE TITLE
Cannot edit exported property when using export_custom with Variant type hint

### ISSUE THREAD

**Issue by Gnumaru** [godotengine/godot#95693]
### Tested versions

- Reproducible in: v4.3.stable.official [77dcf97d8]

### System information

Godot v4.3.stable - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce GTX 1060 6GB (NVIDIA; 31.0.15.3623) - AMD A10-7850K Radeon R7, 12 Compute Cores 4C+8G (4 Threads)

### Issue description

When using the new ```@export_custom``` annotation for untyped properties, the inspector behavior is different if the user uses no type hints or the Variant type hint

![image](https://github.com/user-attachments/assets/e5e40eb9-4fbb-466b-b709-1444a9dbef8e)

The inspector only shows a usable field correctly when not using any type hint. It will show the type hint for the same type of the default value used as initializer for the property in the script.

But when using the ```Variant``` type hint, regardless of the property initializer value, the editor inspector will show a ```"<null>"``` label and the value won't be editable at all. I imagine the desired behavior would be for an usable inspector field to be show regardless when either not using type hints or explicitly using Variant as type hint.

As a side note, a similar thing happens with the data property of json resources, since the data property of a json resource is just a plain variant, it will always show a "<null>" label regardless of the property contents. Please note I'm just pointing out an already existing similar behavior, not reporting a new issue.

![image](https://github.com/user-attachments/assets/5d6ea220-3900-49a0-ace3-c40c0e262671)

### Steps to reproduce

1) in any project, create a new script
1) put the code bellow there
```gdscript
@export_custom(0, '')
var exported_var_without_type_hint = 1.23
@export_custom(0, '')
var exported_var_using_variant_type_hint:Variant = 1.23
```

### Minimal reproduction project (MRP)

[ExportCustomWithVariantBug.zip](https://github.com/user-attachments/files/16644846/ExportCustomWithVariantBug.zip)

### Other notes

Would this [PR](https://github.com/godotengine/godot/pull/89324) fix this?

**Comment by AThousandShips** [comment]
Thank you for reporting! This is a limitation by design rather than a bug, see:
* https://github.com/godotengine/godot-proposals/issues/9368

Consolidating there, see the linked PR for more details

**Comment by Gnumaru** [comment]
Please note that I'm not specifically talking about the functionality described in this [proposal](https://github.com/godotengine/godot-proposals/issues/9368) or this [pull request](https://github.com/godotengine/godot/pull/89324), even though that would be nice.

What I am reporting here is a different thing.

It is that, when using export_custom with a variant without type hints the editor does show an usable field, but when using the Variant type hint it does not show an usable inspector field. I'm not talking here about being able to select a variant type in a drop down like in the mentioned proposal or the pull request, Im just saying that, when you use export_custom with an untyped property it shows an usable field of the same type of the initializer value, wich is good because it is usable in the inspector, but when you use the Variant type hint one would expect the same behavior, an usable inspector field, but instead it shows a misleading ```"<null>"``` label.

In other words, I'm saying that, whatever the behavior is, both usages (with or without a variant type hint) should have the same behavior because an untyped variable is the same thing as a variable with the Variant type hint.

**Comment by dalexeev** [comment]
* Probably related: #94290.
